packages/browser-wallet/src/popup/pages/Setup/PerformRecovery.tsx

```diff
@@ -0,0 +1,103 @@
+import PageHeader from '@popup/shared/PageHeader';
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAtom, useAtomValue } from 'jotai';
+import { useTranslation } from 'react-i18next';
+import { networkConfigurationAtom, seedPhraseAtom } from '@popup/store/settings';
+import { popupMessageHandler } from '@popup/shared/message-handler';
+import { InternalMessageType } from '@concordium/browser-wallet-message-hub';
+import { JsonRpcClient, HttpProvider } from '@concordium/web-sdk';
+import { identitiesAtom, identityProvidersAtom } from '@popup/store/identity';
+import PendingArrows from '@assets/svg/pending-arrows.svg';
+import Button from '@popup/shared/Button';
+import { absoluteRoutes } from '@popup/constants/routes';
+import { BackgroundResponseStatus } from '@shared/utils/types';
+import { getIdentityProviders } from '@popup/shared/utils/wallet-proxy';
+
+// TODO: improve "error state"
+// TODO: Special handling for no identities found
+
+function DisplayRecoveryResult() {
+    const { t } = useTranslation('setup');
+    const navigate = useNavigate();
+    const identities = useAtomValue(identitiesAtom);
+
+    return (
+        <>
+            <div className="onboarding-setup__page-with-header__description">
+                {t('performRecovery.description.after')}
+            </div>
+            <div className="onboarding-setup__recovery__results">
+                {identities.map((identity) => (
+                    <div className="onboarding-setup__recovery__identity">
+                        <p>{identity.name}</p>
+                    </div>
+                ))}
+            </div>
+            <Button onClick={() => navigate(absoluteRoutes.home.account.path)}>{t('continue')}</Button>
+        </>
+    );
+}
+
+export default function EnterRecoveryPhrase() {
+    const { t } = useTranslation('setup');
+    const { jsonRpcUrl } = useAtomValue(networkConfigurationAtom);
+    const masterSeed = useAtomValue(seedPhraseAtom);
+    const [providers, setProviders] = useAtom(identityProvidersAtom);
+    const [result, setResult] = useState<BackgroundResponseStatus>();
+
+    useEffect(() => {
+        getIdentityProviders().then(setProviders);
```

Comment:
If this fails, then recovery never starts, but to the user it looks like it is pending.